hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/CompactionLogEntry.java

```diff
@@ -0,0 +1,179 @@
+      CompactionLogEntryProto proto) {
+    List<FileInfo> inputFileInfo = proto.getInputFileIntoListList().stream()
+        .map(FileInfo::getFromProtobuf)
+        .collect(Collectors.toList());
+
+    List<FileInfo> outputFileInfo = proto.getOutputFileIntoListList().stream()
+        .map(FileInfo::getFromProtobuf)
+        .collect(Collectors.toList());
+
+
+    return new CompactionLogEntry(proto.getDbSequenceNumber(),
+        inputFileInfo, outputFileInfo);
+  }
+
+  public String toEncodedString() {
+    // Encoding is used to deal with \n. Protobuf appends \n after each
+    // parameter. If ByteArray is simply converted to a string or
+    // protobuf.toString(), it will contain \n and will be added to the log.
+    // Which creates a problem while reading compaction logs.
+    // Compaction log lines are read sequentially assuming each line is one
+    // compaction log entry.
+    return Base64.getEncoder().encodeToString(getProtobuf().toByteArray());
+  }
+
+  public static CompactionLogEntry fromEncodedString(String string) {
+    try {
+      byte[] decode = Base64.getDecoder().decode(string);
+      return getFromProtobuf(CompactionLogEntryProto.parseFrom(decode));
+    } catch (InvalidProtocolBufferException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public String toString() {
+    return String.format("dbSequenceNumber: '%s', inputFileInfoList: '%s'," +
+            " outputFileInfoList: '%s',", dbSequenceNumber, inputFileInfoList,
+        outputFileInfoList);
+  }
+
+  public static CompactionLogEntry fromCompactionFiles(
+      long dbSequenceNumber,
+      List<String> inputFiles,
+      List<String> outputFiles
+  ) {
+
+    try (ManagedOptions options = new ManagedOptions();
+         ManagedReadOptions readOptions = new ManagedReadOptions()) {
+      List<FileInfo> inputFileInfos = convertFileInfo(inputFiles, options,
+          readOptions);
+      List<FileInfo> outputFileInfos = convertFileInfo(outputFiles, options,
+          readOptions);
+      return new CompactionLogEntry(dbSequenceNumber, inputFileInfos,
+          outputFileInfos);
+    }
+  }
+
+  private static List<FileInfo> convertFileInfo(
+      List<String> sstFiles,
+      ManagedOptions options,
+      ManagedReadOptions readOptions
+  ) {
+    if (CollectionUtils.isEmpty(sstFiles)) {
+      return Collections.emptyList();
+    }
+
+    final int fileNameOffset = sstFiles.get(0).lastIndexOf("/") + 1;
+    List<FileInfo> response = new ArrayList<>();
+
+    for (String sstFile : sstFiles) {
+      String fileName = sstFile.substring(fileNameOffset,
+          sstFile.length() - SST_FILE_EXTENSION_LENGTH);
+      SstFileReader fileReader = new SstFileReader(options);
+      try {
+        fileReader.open(sstFile);
+        String columnFamily = StringUtils.bytes2String(
+            fileReader.getTableProperties().getColumnFamilyName());
+        SstFileReaderIterator iterator = fileReader.newIterator(readOptions);
```

Comment:
onCompactionCompleted function has the rocks db instance. We can use that to read it from the manifest file getting liveMetadataFile for sst file.

Reply:
You can take look at sst filtering service for this

Reply:
When I looked at it, I couldn't find easy way to get start and end entry. Let me take a look as you are suggesting but I'm unsure if compacted files will be there.

Reply:
For now, let's leave as it is. Will create jira and see if we can use `onFlushCompleted` and read from `LiveFileMetaData` instead of reading through rocksDB iterator.